plans/smlbench/main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"os"
+    "context"
+	"strconv"
+
+	test "github.com/ipfs/testground/plans/smlbench/test"
+	utils "github.com/ipfs/testground/plans/smlbench/utils"
+	iptb "github.com/ipfs/testground/sdk/iptb"
+	"github.com/ipfs/testground/sdk/runtime"
+)
+
+/*
+var testCases = []func(*runtime.RunEnv){
+	test.SimpleAdd,
+	test.SimpleAddGet,
+}
+*/
+
+// Inventory of Tests
+var testCasesSet = [][]utils.SmallBenchmarksTestCase{
+	{
+		&test.SimpleAddTC{},                 // 0
+		&test.SimpleAddTC{1024},             // 1kb
+		&test.SimpleAddTC{64 * 1024},        // 64kb
+		&test.SimpleAddTC{256 * 1024},       // 256kb
+		&test.SimpleAddTC{512 * 1024},       // 512kb
+		&test.SimpleAddTC{1024 * 1024},      // 1mb
+		&test.SimpleAddTC{2 * 1024 * 1024},  // 2mb
+		&test.SimpleAddTC{5 * 1024 * 1024},  // 5mb
+		&test.SimpleAddTC{10 * 1024 * 1024}, // 10mb
+	},
+	{
+		&test.SimpleAddGetTC{},                 // 0
+		&test.SimpleAddGetTC{1024},             // 1kb
+		&test.SimpleAddGetTC{64 * 1024},        // 64kb
+		&test.SimpleAddGetTC{256 * 1024},       // 256kb
+		&test.SimpleAddGetTC{512 * 1024},       // 512kb
+		&test.SimpleAddGetTC{1024 * 1024},      // 1mb
+		&test.SimpleAddGetTC{2 * 1024 * 1024},  // 2mb
+		&test.SimpleAddGetTC{5 * 1024 * 1024},  // 5mb
+		&test.SimpleAddGetTC{10 * 1024 * 1024}, // 10mb
+	},
+}
+
+// TODO:
+//  Testcase abstraction.
+//  Entrypoint demuxing (TEST_CASE_SEQ).
+//  Pipe stdout to intercept messages.
+//  Temporary directory from environment variable.
+//  Error handling -- right now everything panics on failure.
+func main() {
+	// _ = os.Setenv("TEST_PLAN", "smlbenchmarks")
+	// _ = os.Setenv("TEST_BRANCH", "master")
+	// _ = os.Setenv("TEST_TAG", "")
+	// _ = os.Setenv("TEST_RUN", uuid.New().String())
+
+	runenv := runtime.CurrentRunEnv()
+	if runenv.TestCaseSeq < 0 {
+		panic("test case sequence number not set")
+	}
+
+	testCases := testCasesSet[runenv.TestCaseSeq]
+
+	for i, tc := range testCases {
+		_ = os.Setenv("TEST_CASE", tc.Name())
```

Comment:
Hitting:
```
# github.com/ipfs/testground/plans/smlbench
./main.go:67:32: tc.Name undefined (type utils.SmallBenchmarksTestCase has no field or method Name)
```

Although the method is declared. Means that I'm missing some goism here. Will figure this out

Reply:
I added this line to add .Name() to the interface

https://github.com/ipfs/testground/blob/plan-smlbench-single-container/plans/smlbench/utils/testcase.go#L10

My guess is that when it is compiling in the container, it doesn't have the latest code... I wonder if it's pulling from github instead of copying the local source code into the docker image?